File: lib/theme/dariel_theme.dart
import 'package:flutter/material.dart';

class DARIELTheme {
  // Couleurs principales
  static const Color accentRed = Color(0xFFCC1A1A); // Rouge pour les acronymes
  static const Color primaryColor = Color(0xFF664D99); // Indigo
  static const Color secondaryColor = Color(0xFFFAF5FF); // Fond clair
  static const Color cardBackground = Colors.white;
  static const Color textPrimary = Colors.black87;
  static const Color textSecondary = Colors.black54;

  // Dégradés
  static const LinearGradient gradientBackground = LinearGradient(
    colors: [Color(0xFF664D99), Color(0xFFE66699)],
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
  );

  // Espacements
  static const double spacingSmall = 8.0;
  static const double spacingMedium = 16.0;
  static const double spacingLarge = 24.0;

  // Rayons de coins
  static const double cornerRadiusSmall = 12.0;
  static const double cornerRadiusMedium = 16.0;
  static const double cornerRadiusLarge = 24.0;

  // Ombres
  static List<BoxShadow> get shadowSmall => [
        BoxShadow(
          color: Colors.black.withOpacity(0.1),
          blurRadius: 4,
          offset: const Offset(0, 2),
        ),
      ];

  static List<BoxShadow> get shadowMedium => [
        BoxShadow(
          color: Colors.black.withOpacity(0.15),
          blurRadius: 8,
          offset: const Offset(0, 4),
        ),
      ];

  // Polices
  static TextStyle titleFont({double size = 28}) => TextStyle(
        fontSize: size,
        fontWeight: FontWeight.bold,
        fontFamily: 'Roboto',
      );

  static TextStyle subtitleFont({double size = 20}) => TextStyle(
        fontSize: size,
        fontWeight: FontWeight.w600,
        fontFamily: 'Roboto',
      );

  static TextStyle bodyFont({double size = 16}) => TextStyle(
        fontSize: size,
        fontWeight: FontWeight.normal,
      );

  static TextStyle captionFont({double size = 14}) => TextStyle(
        fontSize: size,
        fontWeight: FontWeight.w500,
      );

  // Thème de l'application
  static ThemeData get lightTheme => ThemeData(
        primaryColor: primaryColor,
        colorScheme: ColorScheme.fromSeed(
          seedColor: primaryColor,
          primary: primaryColor,
          secondary: accentRed,
        ),
        scaffoldBackgroundColor: secondaryColor,
        cardTheme: CardTheme(
          color: cardBackground,
          elevation: 4,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(cornerRadiusMedium),
          ),
        ),
        appBarTheme: const AppBarTheme(
          backgroundColor: primaryColor,
          foregroundColor: Colors.white,
          elevation: 0,
        ),
        inputDecorationTheme: InputDecorationTheme(
          filled: true,
          fillColor: Colors.white,
          border: OutlineInputBorder(
            borderRadius: BorderRadius.circular(cornerRadiusSmall),
            borderSide: BorderSide(color: Colors.grey.shade300),
          ),
          enabledBorder: OutlineInputBorder(
            borderRadius: BorderRadius.circular(cornerRadiusSmall),
            borderSide: BorderSide(color: Colors.grey.shade300),
          ),
          focusedBorder: OutlineInputBorder(
            borderRadius: BorderRadius.circular(cornerRadiusSmall),
            borderSide: const BorderSide(color: primaryColor, width: 2),
          ),
        ),
        elevatedButtonTheme: ElevatedButtonThemeData(
          style: ElevatedButton.styleFrom(
            backgroundColor: accentRed,
            foregroundColor: Colors.white,
            padding: const EdgeInsets.symmetric(
              horizontal: spacingLarge,
              vertical: spacingMedium,
            ),
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(cornerRadiusSmall),
            ),
          ),
        ),
        useMaterial3: true,
      );
}

// Extensions pour faciliter l'utilisation du thème
extension DARIELCardStyle on Widget {
  Widget darielCardStyle() {
    return Container(
      padding: const EdgeInsets.all(DARIELTheme.spacingMedium),
      decoration: BoxDecoration(
        color: DARIELTheme.cardBackground,
        borderRadius: BorderRadius.circular(DARIELTheme.cornerRadiusMedium),
        boxShadow: DARIELTheme.shadowMedium,
      ),
      child: this,
    );
  }
}

File: Le_DARIEL/Theme/DARIELTheme.swift
//
//  DARIELTheme.swift
//  Le_DARIEL
//
//  Created by FAEI971 on 23/07/2025.
//

import SwiftUI

struct DARIELTheme {
    // Couleurs principales selon la charte graphique
    static let accentRed = Color(red: 0.8, green: 0.1, blue: 0.1) // Rouge pour les acronymes
    static let primaryColor = Color(red: 0.4, green: 0.3, blue: 0.7) // Indigo
    static let secondaryColor = Color(red: 0.98, green: 0.96, blue: 1.0) // Fond clair
    static let cardBackground = Color.white
    static let textPrimary = Color.primary
    static let textSecondary = Color.secondary
    
    // Dégradés
    static let gradientBackground = LinearGradient(
        colors: [Color(red: 0.4, green: 0.3, blue: 0.7), Color(red: 0.9, green: 0.4, blue: 0.6)],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
    
    // Polices avec taille ajustable
    static func titleFont(size: CGFloat = 28) -> Font {
        .system(size: size, weight: .bold, design: .rounded)
    }
    
    static func subtitleFont(size: CGFloat = 20) -> Font {
        .system(size: size, weight: .semibold, design: .rounded)
    }
    
    static func bodyFont(size: CGFloat = 16) -> Font {
        .system(size: size, weight: .regular, design: .default)
    }
    
    static func captionFont(size: CGFloat = 14) -> Font {
        .system(size: size, weight: .medium, design: .default)
    }
    
    // Espacements
    static let spacingSmall: CGFloat = 8
    static let spacingMedium: CGFloat = 16
    static let spacingLarge: CGFloat = 24
    
    // Rayons de coins
    static let cornerRadiusSmall: CGFloat = 12
    static let cornerRadiusMedium: CGFloat = 16
    static let cornerRadiusLarge: CGFloat = 24
    
    // Ombres
    static let shadowSmall = Shadow(color: .black.opacity(0.1), radius: 4, x: 0, y: 2)
    static let shadowMedium = Shadow(color: .black.opacity(0.15), radius: 8, x: 0, y: 4)
    
    struct Shadow {
        let color: Color
        let radius: CGFloat
        let x: CGFloat
        let y: CGFloat
    }
}

// MARK: - View Extensions pour le style DARIEL

extension View {
    func darielCardStyle() -> some View {
        self
            .padding(DARIELTheme.spacingMedium)
            .background(DARIELTheme.cardBackground)
            .cornerRadius(DARIELTheme.cornerRadiusMedium)
            .shadow(
                color: DARIELTheme.shadowMedium.color,
                radius: DARIELTheme.shadowMedium.radius,
                x: DARIELTheme.shadowMedium.x,
                y: DARIELTheme.shadowMedium.y
            )
    }
    
    func darielAcronymStyle() -> some View {
        self
            .foregroundColor(DARIELTheme.accentRed)
            .font(DARIELTheme.titleFont())
    }
}

// MARK: - Helper pour formater la définition selon la charte
extension String {
    func formattedDefinition() -> AttributedString {
        var attributed = AttributedString(self)
        
        // Première lettre en majuscule et rouge
        if let firstChar = self.first,
           let firstCharRange = attributed.range(of: String(firstChar)) {
            attributed[firstCharRange].foregroundColor = DARIELTheme.accentRed
            attributed[firstCharRange].font = .system(size: 18, weight: .bold)
        }
        
        return attributed
    }
}





